metrics/web/lib/features/common/presentation/routes/route_generator.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:metrics/features/auth/presentation/pages/login_page.dart';
+import 'package:metrics/features/dashboard/presentation/pages/dashboard_page.dart';
+
+/// Responsible for generating routes.
+class RouteGenerator {
+  /// Generates a route for the given route [settings] and a user authentication
+  /// status, determined by a [isLoggedIn] value.
+  static MaterialPageRoute generateRoute(
+      {RouteSettings settings, bool isLoggedIn}) {
+    switch (settings.name) {
+      case '/dashboard':
```

Comment:
I think it would be great to extract route names to the constants. What do you think about this?